include/seqan3/io/alignment_file/input.hpp

```diff
@@ -0,0 +1,1323 @@
+                      return true;
+                  }(),
+                  "You selected a field that is not valid for aligment files, please refer to the documentation "
+                  "of alignment_file_input::field_ids for the accepted values.");
+
+    //!\brief The type of the record, a specialisation of seqan3::record; acts as a tuple of the selected field types.
+    using record_type        = record<detail::select_types_with_ids_t<field_types, field_ids, selected_field_ids>,
+                                      selected_field_ids>;
+    //!\}
+
+    /*!\name Range associated types
+     * \brief The types necessary to facilitate the behaviour of an input range (used in record-wise reading).
+     * \{
+     */
+    //!\brief The value_type is the \ref record_type.
+    using value_type        = record_type;
+    //!\brief The reference type.
+    using reference         = record_type &;
+    //!\brief The const_reference type is void, because files are not const-iterable.
+    using const_reference   = void;
+    //!\brief An unsigned integer type, usually std::size_t.
+    using size_type         = size_t;
+    //!\brief A signed integer type, usually std::ptrdiff_t.
+    using difference_type   = std::make_signed_t<size_t>;
+    //!\brief The iterator type of this view (an input iterator).
+    using iterator          = detail::in_file_iterator<alignment_file_input>;
+    //!\brief The const iterator type is void, because files are not const-iterable.
+    using const_iterator    = void;
+    //!\brief The type returned by end().
+    using sentinel          = std::ranges::default_sentinel_t;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Default constructor is explicitly deleted, you need to give a stream or file name.
+    alignment_file_input() = delete;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
+    alignment_file_input(alignment_file_input const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
+    alignment_file_input & operator=(alignment_file_input const &) = delete;
+    //!\brief Move construction is defaulted.
+    alignment_file_input(alignment_file_input &&) = default;
+    //!\brief Move assignment is defaulted.
+    alignment_file_input & operator=(alignment_file_input &&) = default;
+    //!\brief Destructor is defaulted.
+    ~alignment_file_input() = default;
+
+    /*!\brief Construct from filename.
+     * \param[in] filename    Path to the file you wish to open.
+     * \param[in] fields_tag  A seqan3::fields tag. [optional]
+     * \throws seqan3::file_open_error If the file could not be opened, e.g. non-existant, non-readable, unknown format.
+     *
+     * \details
+     *
+     * In addition to the file name, you may specify a custom seqan3::fields type which may be easier than
```

Comment:
but you give a value of a custom seqan3::fields?

Reply:
I give him a field`s` type of field values. (e.g. `seqan3::fields<seqan3::field::SEQ>{}`)
I added this example in the comment

Reply:
and I will call it `fields object` instead of type.